scripts/cog_special_methods.py

```diff
@@ -0,0 +1,114 @@
+# SPDX-License-Identifier: MIT
+
+
+"""Add all special cog methods to the full list contained in disnake.__main__.
+
+This script does not run automatically and special methods that are removed from cogs
+must also be removed from the list manually.
+"""
+from __future__ import annotations
+
+import inspect
+import os
+import re
+import sys
+import textwrap
+from typing import Dict, Union
+
+_RE = r""" = (?:"|'){3}\n(.+?)\n(?:'|"){3}"""
+
+
+def remove_annotations_from_sig(sig: inspect.Signature) -> inspect.Signature:
+    return sig.replace(
+        parameters=[
+            param.replace(annotation=inspect.Parameter.empty) for param in sig.parameters.values()
+        ],
+        return_annotation=inspect.Signature.empty,
+    )
+
+
+def get_cog_special_methods() -> Dict[str, str]:
+
+    from disnake.ext.commands import Cog
+
+    _sentinel = object()
+    result: Dict[str, str] = {}
+    for name, attr in sorted((name, getattr(Cog, name)) for name in dir(Cog)):
+        if getattr(attr, "__cog_special_method__", _sentinel) is _sentinel:
+            continue
+
+        async_ = "async " if inspect.iscoroutinefunction(attr) else ""
+        sig = inspect.signature(attr)
+        sig = remove_annotations_from_sig(sig)
+        result[name] = f"{async_}def {name}{str(sig)}:"
+
+    return result
+
+
+def format_sig(name: str, sig: str, body: str = "...") -> str:
+    body = textwrap.indent(textwrap.dedent(body), " " * 4)
+    return f"{sig}\n{body}\n"
+
+
+def merge_methods(contents: str, methods: Dict[str, str]) -> str:
+
+    for name, sig in methods.items():
+
+        if f"def {name}" in contents:
+            # check that the signature is the same
+            missing_async = False
+            if sig not in contents or (
+                missing_async := (not sig.startswith("async") and ("async " + sig) in contents)
```

Comment:
```suggestion
                missing_async := (not sig.startswith("async ") and re.search(rf"async {sig}\b", contents))
```

This entire block could be simplified by matching something like `^\s*{sig}\b`, as far as I can tell.

Unrelated to that, this code doesn't account for special methods that *can* be coroutine functions, like `bot_check` (where this script is currently raising the `RuntimeError` below). I'm wondering if this part is worth the hassle at all...

Reply:
The script was just to help generate the methods initially, it doesn't have to work perfectly :)

Reply:
Honestly I'd rather remove it than make it work properly

Reply:
> Honestly I'd rather remove it than make it work properly

Yeah, agreed. I'd prefer maintaining the template manually (which should be very rare), instead of having to maintain an entire generator script.